relay-general/src/pii/processor.rs

```diff
@@ -43,7 +43,7 @@ impl<'a> PiiProcessor<'a> {
                 #[allow(clippy::needless_option_as_deref)]
                 for rule in rules {
                     let reborrowed_value = value.as_deref_mut();
-                    apply_rule_to_value(meta, rule, state.path().key(), reborrowed_value)?;
+                    apply_rule_to_value(&mut meta, rule, state.path().key(), reborrowed_value)?;
```

Comment:
In case we keep the optional: It should be possible to pass an `Option<&mut Meta>` here by doing the same as for `value`, that is `let reborrowed_meta = meta.as_deref_mut();`